Controller/Midpage/AbstractAction.php

```diff
@@ -0,0 +1,84 @@
+<?php
+namespace Billink\Billink\Controller\Midpage;
+
+use Billink\Billink\Gateway\Helper\TransactionManager;
+use Billink\Billink\Model\Payment\Session;
+use Magento\Checkout\Model\Session as CheckoutSession;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\App\Action\HttpGetActionInterface;
+use Magento\Framework\App\RequestInterface;
+use Magento\Framework\Controller\Result\RedirectFactory;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Message\ManagerInterface;
+use Magento\Payment\Gateway\Command\CommandPoolInterface;
+use Magento\Payment\Gateway\Data\PaymentDataObjectFactory;
+use Magento\Sales\Api\OrderRepositoryInterface;
+use Psr\Log\LoggerInterface;
+
+abstract class AbstractAction implements HttpGetActionInterface
+{
+    protected RequestInterface $request;
```

Comment:
if we are on PHP8+, we can use constructor properties (and iirc from 8.1 on, we can even do readonly properties, saves on clutter

Reply:
@bakkerpj  This one is a head-scratcher for me here.
Mostly I thought that this should be PHP7 compatible, but now I see that some functions have `: static` return, which is PHP8. I don't see other places with PHP8-specific, so it almost looks like the extension is still compatible with 7, but not quite.
And the compatibility is the reason I have not used those. So either we drop PHP7 completely (and probably update composer) or we should fix `: static` return.

Reply:
@DarthFly We have explicitly stated: `Versions 2.0.0 and later of this module require Magento 2.4+ and PHP 8.1+.`. So feel free to drop the PHP7 compatibility.

Reply:
@DarthFly And yes, it is a good idea to enforce this through requirements in the `composer.json`.